File: api/src/models/auth/enums.rs
use crate::{
    define_enum,
    generated::auth_schema::sql_types::{ApiKeyPermission, UserStatus, WorkspaceRole},
};
use std::io::Write;

define_enum!(
    WorkspaceRoleEnum,
    WorkspaceRole,
    [User, Manager, Admin],
    WithConversion
);
define_enum!(
    UserStatusEnum,
    UserStatus,
    [Active, Suspended],
    WithConversion
);
define_enum!(
    ApiKeyPermissionEnum,
    ApiKeyPermission,
    [ReadOnly, WriteOnly, ReadWrite],
    WithConversion
);

File: prototype/prototype/models/enums.py
"""Enums."""

from enum import StrEnum


class UserRole(StrEnum):
    """User role."""

    USER = "USER"
    ADMIN = "ADMIN"
    SYSADMIN = "SYSADMIN"


class ApiKeyPermission(StrEnum):
    """API Key Permission Enum"""

    READ_ONLY = "READ_ONLY"
    WRITE_ONLY = "WRITE_ONLY"
    READ_WRITE = "READ_WRITE"


class WorkspaceRole(StrEnum):
    """Workspace role."""

    USER = "USER"
    MANAGER = "MANAGER"
    ADMIN = "ADMIN"
